src/Basement/3DBasementHT.f90

```diff
@@ -9634,7 +9641,7 @@ SUBROUTINE CalcTearth(IEXT,JEXT,DZ,DZP,TG,CVG)
 
 
 !*** DECLARATIONS:
-     REAL(r64) A(50), B(50), C(50), R(50), X(50), ALB, ALBEDO(2),             &
+     REAL(r64) A(100), B(100), C(100), R(100), X(100), ALB, ALBEDO(2),             &
```

Comment:
Actual fix is here

ZFACE is ZFACE(-35:100), positive indicates Below ground (NZBG).
A, B, C,R and X are used in the 1 to NZBG range, so they should be 100 elements not 50 (Note: they are 1-indexed)

Reply:
I think the -35:100 means there are 136 array elements?

https://www-eio.upc.es/lceio/manuals/Fortran95-manual.pdf

    All the following type declaration statements are legal:
     real, dimension (-10:8) :: a1 ! 1-dim array with 19 elements
     integer, dimension (-3:3, -20:0, 1:2, 6, 2, 5:6, 2) :: grid1 ! 7-dim array
    The number of elements of the integer array grid1 is 7 x 21 x 2 x 6 x 2 x 2 x 2 = 14112.

Reply:
But NZBG is a user input and I think the idd is suggesting a range of 10 to 35 for that input, so that's where the 50 came from? There is no upper limit on that field so a user could enter 150 and crash the program again.

    GroundHeatTransfer:Basement:ManualGrid,
        \memo Manual Grid only necessary using manual gridding (not recommended)
           N1, \field NX: Number of cells in the X direction: 20]
              \minimum 1
              \required-field
           N2, \field NY: Number of cells in the Y direction: 20]
              \minimum 1
              \required-field
           N3, \field NZAG: Number of cells in the Z direction. above grade: 4 Always]
              \minimum 1
              \required-field
           N4, \field NZBG: Number of cells in Z direction. below grade: 10-35]
              \minimum 1
              \required-field

     NumNums=GetNumObjectsFound('ManualGrid')
     IF (NumNums > 0) THEN
       CALL GetObjectItem('ManualGrid',NUM,AlphArray,NumAlphas,NumArray,NumNums,IOSTAT)
       NX   =NumArray(1)
       NY   =NumArray(2)
       NZAG =NumArray(3)
       NZBG =NumArray(4)

Reply:
ManualGrid is broken, there is ZERO way to use it (AUTOGRID is set to TRUE and it's never ressetable to FALSE): https://github.com/NREL/EnergyPlus/blob/acda84e38db4e8aecbc178cb970055d3721ee321/src/Basement/3DBasementHT.f90#L999

I brought it up with @Myoldmopar, not sure if it was worth fixing (the Documentation explicitly says this is discouraged anyways).

The EquivAutoGrid BaseDepth is the one that determines the NZBG, and I agree I should probably do a bounds check there.

---

There are 136 elements in ZFACE, 35 < 0, the `0`, and `100`> 0 (1 to 100).

If you look closely at the routine, you'll see the A, B etc arrays are only used in the 1 to NZBG range, so it needs 100 elements.